### DISCUSSION TITLE
Change Detection LXC Installs but Browserless service cannot start

### DISCUSSION THREAD

**Discussion by pieslinger** [tteck/Proxmox#2262] [Help Needed]
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

After an update, my change detection lxc stopped working for all watches that used the Playwright fetch method. I erased the container and started over. When I run
```
systemctl status browserless.service
```
I  see:
```
Dec 27 23:47:29 changedetection browserless[1175]: Unhandled Rejection at: Promise {
Dec 27 23:47:29 changedetection browserless[1175]:   <rejected> Error: Couldn't load route "/content" due to missing browser of "CDPChromium"
Dec 27 23:47:29 changedetection browserless[1175]:       at file:///opt/browserless/build/browserless.js:148:23
Dec 27 23:47:29 changedetection browserless[1175]:       at Array.forEach (<anonymous>)
Dec 27 23:47:29 changedetection browserless[1175]:       at Browserless.start (file:///opt/browserless/build/browserless.js:145:38)
Dec 27 23:47:29 changedetection browserless[1175]: } reason: Error: Couldn't load route "/content" due to missing browser of "CDPChromium"
Dec 27 23:47:29 changedetection browserless[1175]:     at file:///opt/browserless/build/browserless.js:148:23
Dec 27 23:47:29 changedetection browserless[1175]:     at Array.forEach (<anonymous>)
Dec 27 23:47:29 changedetection browserless[1175]:     at Browserless.start (file:///opt/browserless/build/browserless.js:145:38)
Dec 27 23:47:29 changedetection browserless[1175]: 2023-12-28T06:47:29.207Z browserless:index.js Process is finished, exiting
```
I installed the browsers manually with
```
npm run install:dev
``` 
and then I can manually start browerless with the start.sh script in opt/browserless and then changedetection and the watches that rely on the playwright work as they should.

However, even though after the browsers are installed and I can manually start browserless, the service reports the same error on start. 

### Which Linux distribution are you employing?

Debian 12

### Please provide detailed steps to reproduce the issue.

Use install script from website:

bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/changedetection.sh)"

Setup a new watch with the the playwright fetch method and you should always see this error:
"WebSocket error: connect ECONNREFUSED 127.0.0.1:3000 "

then if you check the browserless service, you should see what I recorded above.

**Comment by tteck** [comment]
My primary focus revolves around the development of application LXCs, and I'm not specialized in providing support for operational issues that may arise with these applications. 

Admittedly, I know nothing about how this application works, never used it.

It suggests that the browserless service is encountering an error when trying to load the specified route "/content" because it can't find the required browser ("CDPChromium"). 

I'm going to send this issue to Discussions. Hopefully someone will advise with a solution. 

The browserless service is running.

```
root@changedetection:~# systemctl status browserless.service
* browserless.service - browserless service
     Loaded: loaded (/etc/systemd/system/browserless.service; enabled; preset: enabled)
     Active: active (running) since Thu 2023-12-28 05:43:39 EST; 2min 46s ago
   Main PID: 8337 (start.sh)
      Tasks: 2 (limit: 12977)
     Memory: 85.4M
        CPU: 4.033s
     CGroup: /system.slice/browserless.service
             |-8337 /bin/bash /opt/browserless/start.sh
             `-8339 Xvfb :99 -screen 0 1920x1080x16 -nolisten tcp -nolisten unix

Dec 28 05:43:48 changedetection browserless[8341]: Unhandled Rejection at: Promise {
Dec 28 05:43:48 changedetection browserless[8341]:   <rejected> Error: Couldn't load route "/content" due to missing br>
Dec 28 05:43:48 changedetection browserless[8341]:       at file:///opt/browserless/build/browserless.js:148:23
Dec 28 05:43:48 changedetection browserless[8341]:       at Array.forEach (<anonymous>)
Dec 28 05:43:48 changedetection browserless[8341]:       at Browserless.start (file:///opt/browserless/build/browserles>
Dec 28 05:43:48 changedetection browserless[8341]: } reason: Error: Couldn't load route "/content" due to missing brows>
Dec 28 05:43:48 changedetection browserless[8341]:     at file:///opt/browserless/build/browserless.js:148:23
Dec 28 05:43:48 changedetection browserless[8341]:     at Array.forEach (<anonymous>)
Dec 28 05:43:48 changedetection browserless[8341]:     at Browserless.start (file:///opt/browserless/build/browserless.>
Dec 28 05:43:48 changedetection browserless[8341]: 2023-12-28T10:43:48.665Z browserless:index.js Process is finished, e>
```

**Reply by pieslinger** [reply]
Makes sense.

**Comment by lukyjay** [comment]
Hi the problem is they have recently released a new major version of Browserless. Install the old version instead.

For an existing install, just run this:
`mv /opt/browserless /opt/browserless_old`
`git clone -b v1 https://github.com/browserless/chrome /opt/browserless`
`npm install --prefix /opt/browserless`
`npm run build --prefix /opt/browserless`
`npm prune production --prefix /opt/browserless`

Then restart the browserless service with service browserless restart. It works now

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/commit/a9b44432c4ff5810582de9bc11c03e7c8e346d83

**Reply by pieslinger** [reply]
Thanks!

**Reply by lukyjay** [reply]
@tteck @pieslinger moving forward we should find a way to get browserless V2 working. There may be security vulnerabilities in older versions

**Reply by Lockie85** [reply]
@lukyjay did you find a way to resolve this issue with latest? I'm not to sure, but I think I might have the same issue: https://github.com/tteck/Proxmox/discussions/3822
